### ISSUE TITLE
bug: waitForTransaction through error multiple times

### ISSUE THREAD

**Issue by atomicDev03** [wagmi-dev/wagmi#3123]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.4.3

### Current Behavior

When calling the waitForTransaction function, it sometimes throws an error multiple times. When I put waitForTransaction inside a try/catch block, it catches the error the first time but then continues to throw it again. In another issue, some users mentioned having this problem with both wagmi and viem (#2530,  #2505). One of your developers responded, saying the error is related to the public RPC. That's a valid point, but why does it throw the error multiple times? I believe this behavior is due to the asynchronous nature of the function. Shouldn't it wait for all asynchronous functions to run and then either throw an error or return a result?

### Expected Behavior

My expected behavior is for the function to either return the transaction or throw an error only once.

### Steps To Reproduce

[link to reproduce](https://github.com/wagmi-dev/wagmi/issues/2530#:~:text=https%3A//stackblitz.com/edit/new%2Dwagmi%2Dq5cuif%3Ffile%3Dsrc%252Fcomponents%252FNetworkSwitcher.tsx) ( copy from #2530) Please note that this behavior occurs randomly, so the issue might or might not be reproduced.

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

https://github.com/wagmi-dev/wagmi/issues/2530#:~:text=https%3A//stackblitz.com/edit/new%2Dwagmi%2Dq5cuif%3Ffile%3Dsrc%252Fcomponents%252FNetworkSwitcher.tsx

**Comment by jxom** [comment]
Public RPCs have aggressive load balancing which causes some functionality that relies on synchronous behaviour like this to break. 

Public RPCs should not be used in production anyway. Please try this again with an RPC like Alchemy, Ankr, QuickNode or Infura.

**Comment by PolEpie** [comment]
Hi ! I still got this error with ankr rpc on fantom, is there a way to slower the refresh rate or something to avoid this ?

**Comment by atomicDev03** [comment]
I still got this error with priced rpc.
I have a question. How to enforce wagmi to use injected wallet( window.ethereum) for read action?
@jxom
